test_trade_execution_comprehensive.py

```
@@ -0,0 +1,675 @@
+            # Should fallback to simulation with notice
+            assert result["success"] is True
+            assert result.get("simulation_fallback") is True
+            assert "simulation mode due to missing exchange credentials" in result.get("notice", "")
+
+            print("[OK] Fix #3: Intelligent fallback to simulation works")
+
+    @pytest.mark.asyncio
+    async def test_fix_4_chat_respects_user_preference(self):
+        """Test Fix #4: Chat engine respects user's simulation preference"""
+
+        chat_engine = EnhancedAIChatEngine()
```

Ruff (0.12.2)

Undefined name `EnhancedAIChatEngine`

(F821)